process/transaction/shardProcess.go

```diff
@@ -633,34 +637,68 @@ func (txProc *txProcessor) addFeeAndValueToDest(acntDst state.UserAccountHandler
 
 func (txProc *txProcessor) processRelayedTxV3(
 	tx *transaction.Transaction,
-	relayerAcnt, acntDst state.UserAccountHandler,
+	relayerAcnt state.UserAccountHandler,
 ) (vmcommon.ReturnCode, error) {
 	if !txProc.enableEpochsHandler.IsFlagEnabled(common.RelayedTransactionsV3Flag) {
 		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrRelayedTxV3Disabled)
 	}
-	if tx.GetValue().Cmp(big.NewInt(0)) != 0 {
-		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrRelayedTxV3ZeroVal)
+	if check.IfNil(relayerAcnt) {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrNilRelayerAccount)
+	}
+	err := txProc.relayedTxV3Processor.CheckRelayedTx(tx)
+	if err != nil {
+		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, err)
 	}
 
-	userTx := tx.GetInnerTransaction()
-	if !bytes.Equal(tx.RcvAddr, userTx.SndAddr) {
-		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrRelayedTxV3BeneficiaryDoesNotMatchReceiver)
+	// process fees on both relayer and sender
+	relayerFee, totalFee := txProc.relayedTxV3Processor.ComputeRelayedTxFees(tx)
+	err = txProc.processTxAtRelayer(relayerAcnt, totalFee, relayerFee, tx)
+	if err != nil {
+		return 0, err
 	}
-	if len(userTx.RelayerAddr) == 0 {
-		return vmcommon.UserError, txProc.executingFailedTransaction(tx, relayerAcnt, process.ErrRelayedTxV3EmptyRelayer)
+
+	innerTxs := tx.GetInnerTransactions()
+
+	var innerTxRetCode vmcommon.ReturnCode
+	var innerTxErr error
+	executedUserTxs := make([]*transaction.Transaction, 0)
+	for _, innerTx := range innerTxs {
+		innerTxRetCode, innerTxErr = txProc.finishExecutionOfInnerTx(tx, innerTx)
+		if innerTxErr != nil || innerTxRetCode != vmcommon.Ok {
+			continue
+		}
+
+		executedUserTxs = append(executedUserTxs, innerTx)
```

Comment:
we have to check whether an additional logEvent would be needed here - I think it is better to have that innerTX was executed.

Reply:
we do have a SCR. do we also a logEvent?

Reply:
now I understand.. it would make sense to also have events for the succeeded inner txs

Reply:
added in the next pr :pray: